judgels-frontends/raphael/src/routes/contests/contests/single/clarifications/ContestClarificationsPage/ContestClarificationsPage.tsx

```diff
@@ -45,20 +61,36 @@ class ContestClarificationsPage extends React.Component<
 
   state: ContestClarificationsPageState = {};
 
+  constructor(props) {
+    super(props);
+
+    const queries = parse(this.props.location.search);
+    const status = queries.status as string;
+
+    this.state = { filter: { status } };
+  }
+
   render() {
     return (
       <ContentCard>
         <h3>Clarifications</h3>
         <hr />
         {this.renderCreateDialog()}
+        {this.renderFilterWidget()}
+        <div className="clearfix" />
         {this.renderClarifications()}
         {this.renderPagination()}
       </ContentCard>
     );
   }
 
-  private refreshClarifications = async (page?: number) => {
-    const response = await this.props.onGetClarifications(this.props.contest.jid, this.props.statementLanguage, page);
+  private refreshClarifications = async (status?: string, page?: number) => {
```

Comment:
Seems like there is a bug. Steps:

- As a contestant, ask a clarification.
- As a supervisor open clarifications tab.
- Filter ASKED.
- Answer the clarification.
- After that all clarifications will be shown. But the filter is still ASKED.

Can we fix this?

Reply:
ow got it, will check and try find bext experience for this.